### PR TITLE
Sdt chapter5

### PR DESCRIPTION
**Pull request by SandyThomson**

### CODE REVIEW

File: answers/chapter5.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+Q1
+
+       double array[] = new double[12];
+       double[] array = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10.01, 11.02, 12.03 };
+
```
Comment (SimonGross) [inline answers/chapter5.md:5]: I think it was asking for two ways to declare an array, not initialize it with values (i.e. like in Question 2). In that case the second line should be:

``` java
double[] array = new double[12];
```

File: answers/chapter5.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		See Encode(Test).java
+		
+Q8
+
+		No
+
+Q9
+
+		y = (x < 0) ? 10 : 20;
+
```
Comment (SimonGross) [inline answers/chapter5.md:40]: You don't technically need the brackets, but they could be useful for readability.

File: answers/chapter5.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+Q12
+
+		>>>
+		
+		It shifts bits right, but zerofills from the left
+		
+Q13
+
+		// TODO
+
```
Comment (SimonGross) [inline answers/chapter5.md:59]: Tsk, tsk

File: src/com/selftest/DoubleAverage.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+		int count = 0;
+		double sum = 0;
+
+		for (double val : list) {
+			sum += val;
+			count++;
+		}
+
+		if (count > 0) {
+			return sum / count;
```
Comment (SimonGross) [inline src/com/selftest/DoubleAverage.java:16]: You can achieve the same effect by initializing `count` to `list.length` and not incrementing it.

File: src/com/selftest/StringArraySorter.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+		int size = strings.length;
+
+		for (int a = 1; a < size; a++) {
+			for (int b = size - 1; b >= a; b--) {
+
+				// Start off by ignoring case, and then only factor in case if
+				// it might make a difference
+				int compareToResult = strings[b - 1]
+						.compareToIgnoreCase(strings[b]);
```
Comment (SimonGross) [inline src/com/selftest/StringArraySorter.java:15]: :+1: for taking upper/lower case into account. 

The question only asked us to sort the Strings and didn't give criteria for sorting and whether case mattered, so either way is valid imo.

File: test/com/selftest/EncodeTest.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import org.junit.Test;
+
+public class EncodeTest {
+
+	@Test
+	public void testEncodeDecodeLongerString() throws Exception {
+		
+		Encode encode = new Encode();
+		
+		encode.setKey("boobs");
```
Comment (SimonGross) [inline test/com/selftest/EncodeTest.java:13]: Very minor, but the question did ask that the key was 8 characters.

File: answers/chapter5.md (hunk 1 above)
Comment (SandyThomson) [inline answers/chapter5.md:5]: Yeah. I just wanted to make things more different.

File: src/com/selftest/DoubleAverage.java (hunk 4 above)
Comment (SandyThomson) [inline src/com/selftest/DoubleAverage.java:16]: Yeah. I realised my mistake when I started to review other peoples work.

File: src/com/selftest/StringArraySorter.java (hunk 5 above)
Comment (SandyThomson) [inline src/com/selftest/StringArraySorter.java:15]: +1

File: test/com/selftest/EncodeTest.java (hunk 6 above)
Comment (SandyThomson) [inline test/com/selftest/EncodeTest.java:13]: _Totally_ missed that... sigh.

Comment (codavex) [inline test/com/selftest/EncodeTest.java:13]: booooobs

**Pull request closed without merging**